flycheck-elsa.el

```diff
@@ -46,28 +46,119 @@
+flycheck.
+
+If value is a list of strings, it will be passed unchanged as the
+`:command' argument to `flycheck-define-command-checker'."
+  :group 'flycheck-elsa
+  :type '(choice
+          (const :tag "Cask" 'cask)
+          (const :tag "Emacs" 'host)
+          (repeat 'string)))
 
-We require that the project is managed by Cask and that Elsa is
-listed as a dependency."
-  (when (and (buffer-file-name)
-             (not (seq-find (lambda (f) (string-match-p f (buffer-file-name)))
-                            flycheck-elsa-ignored-files-regexps)))
-    (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
-      (let ((bundle (cask-initialize (file-name-directory cask-file))))
-        (cask-find-dependency bundle 'elsa)))))
+(defun flycheck-elsa--filename-ignored-p (filename)
+  "Return non-nil if FILENAME is matched by one of `flycheck-elsa-ignored-files-regexps'."
+  (seq-find
+   (lambda (regexp) (string-match-p regexp filename))
+   flycheck-elsa-ignored-files-regexps))
+
+(defun flycheck-elsa--enable-p ()
+  "Return non-nil if we can enable Elsa in current buffer."
+  (if (flycheck-elsa-cask-p)
+      (when (and (buffer-file-name)
+                 (not (flycheck-elsa--filename-ignored-p (buffer-file-name))))
+        (when-let ((cask-file (locate-dominating-file (buffer-file-name) "Cask")))
+          (let ((bundle (cask-initialize (file-name-directory cask-file))))
+            (cask-find-dependency bundle 'elsa))))
+    (if (buffer-file-name)
+        (not (flycheck-elsa--filename-ignored-p (buffer-file-name)))
+      t)))
 
 (defun flycheck-elsa--working-directory (&rest _)
   "Return the working directory where the checker should run."
   (if (buffer-file-name)
-      (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
-        (file-name-directory file))
+      (if (flycheck-elsa-cask-p)
+          (when-let (file (locate-dominating-file (buffer-file-name) "Cask"))
+            (file-name-directory file))
+        (file-name-directory (cdr (project-current))))
     default-directory))
 
+(defun flycheck-elsa--cask-command ()
+  "Return a list of strings to start Elsa within a Cask project."
+  '("cask" "exec" "elsa"))
+
+(defun flycheck-elsa--elsa-dependency-directory (library-name)
+  "Return the directory containing LIBRARY-NAME."
+  (file-name-directory (find-library-name library-name)))
+
+(defun flycheck-elsa--elsa-directory ()
+  "Return the directory where Elsa is installed."
+  (flycheck-elsa--elsa-dependency-directory "elsa"))
+
+(defun flycheck-elsa--elsa-dependencies ()
+  "Return a list of library names, as strings, that Elsa depend upon.
+The list does not contain Emacs itself."
+  (let* ((default-directory (flycheck-elsa--elsa-directory))
+         (package-description (with-temp-buffer
+                                (dired-mode) ;; required by `package-dir-info'
+                                (package-dir-info)))
```

Comment:
This works with single-file packages also?

Reply:
yes it does, but why do you care? We are looking for Elsa dependencies in the Elsa package.

Reply:
You're right.  I thought this was to add the user's (target) package dependencies to the load path.  We also need to do that.

I only skimmed the code, sorry for the noise.